Order confirmations for Printful products show "free shipping"

On the pundi-x.ogn.app shop, which is integrated with Printful, all of the orders that have been made are showing "free shipping". We should be using the Printful shipping API and charging the customer for shipping, so that the merchant will not have to cover shipping costs.

<img width="440" alt="Screenshot 2020-09-24 at 1 14 05 AM" src="https://user-images.githubusercontent.com/46481405/94046422-412c4a00-fe03-11ea-8134-ed75c19cca4b.png">

<img width="280" alt="Screenshot 2020-09-24 at 1 13 54 AM" src="https://user-images.githubusercontent.com/46481405/94046520-5c975500-fe03-11ea-855b-f199b7e4d6f9.png">

Not positive this is related but I see some errors like this in the backend logs:
```
[ERROR] dshop.app: TypeError: Cannot read property 'address1' of undefined
    at fetchShippingEstimate (/app/backend/utils/printful.js:149:27)
    at router.post (/app/backend/routes/printful.js:63:26)
```

--- a/shop/src/pages/checkout/Shipping.js
+++ b/shop/src/pages/checkout/Shipping.js
@@ -19,7 +19,11 @@ function isActive(zone, cart) {
 const CheckoutShipping = () => {
   const { config } = useConfig()
   const [{ cart }, dispatch] = useStateValue()
-  const { shippingZones, loading } = useFlattenedShippingZones()
+  const {
+    shippingZones,
+    loading,
+    error: shippingZonesError
+  } = useFlattenedShippingZones()
   const currencyOpts = useCurrencyOpts()
 
   const country = get(cart, 'userInfo.country')
@@ -30,7 +34,11 @@ const CheckoutShipping = () => {
   const filteredShippingZones = shippingZones.filter(
     (zone) => (zone.countries || []).indexOf(countryCode) >= 0
   )
-  if (!filteredShippingZones.length && defaultShippingZone) {
+  if (
+    !shippingZonesError &&
+    !filteredShippingZones.length &&
+    defaultShippingZone
+  ) {
     filteredShippingZones.push(defaultShippingZone)
   }
 
@@ -104,10 +112,11 @@ const CheckoutShipping = () => {
               Loading shipping costs...
             </fbt>
           </div>
-        ) : !filteredShippingZones.length ? (
+        ) : shippingZonesError || !filteredShippingZones.length ? (
           <div className="p-3">
             <fbt desc="checkout.shipping.loadError">
-              Sorry, there was an error calculating shipping costs.
+              Sorry, there was an error calculating shipping costs. Try
+              refreshing the page.
             </fbt>
           </div>
         ) : (

--- a/shop/src/utils/useShippingZones.js
+++ b/shop/src/utils/useShippingZones.js
@@ -1,4 +1,5 @@
 import { useEffect, useState } from 'react'
+import get from 'lodash/get'
 
 import { useStateValue } from 'data/state'
 
@@ -8,10 +9,26 @@ function useShippingZones() {
   const { config } = useConfig()
   const [{ shippingZones, cart, reload }, dispatch] = useStateValue()
   const [loading, setLoading] = useState(true)
+  const [configLoading, setConfigLoading] = useState(true)
   const [error, setError] = useState(false)
+
+  const userInfo = get(cart, 'userInfo')
+  const items = get(cart, 'items')
+
+  useEffect(() => {
+    if (!configLoading || !userInfo || !items) {
+      return
+    }
+
+    setConfigLoading(false)
+  }, [userInfo, items, configLoading])
 
   useEffect(() => {
     async function fetchShippingZones() {
+      if (configLoading) {
+        return
+      }
+
       setLoading(true)
       let zones = []
       try {
@@ -50,7 +67,7 @@ function useShippingZones() {
     }
 
     fetchShippingZones()
-  }, [reload.shippingZones])
+  }, [configLoading, reload.shippingZones])
 
   return { shippingZones, loading, error }
 }

